### PR TITLE
[feature](hive)add read of the hive table textfile format array type

### PR DESCRIPTION
**Pull request by hubgeter**
## Proposed changes

Issue Number: close #xxx

1. All base types in Hive are supported except for binary .
Create table and insert data in hive:
```sql
CREATE TABLE ywtest.test (
  a ARRAY<TINYINT>,
  b ARRAY<SMALLINT>,
  c ARRAY<INT>,
  d ARRAY<BIGINT>,
  e ARRAY<BOOLEAN>,
  f ARRAY<FLOAT>,
  g ARRAY<DOUBLE>,
  h ARRAY<STRING>,
  i ARRAY<TIMESTAMP>,
  j ARRAY<DATE>
)
stored as textfile;

INSERT INTO ywtest.test VALUES
(
    ARRAY(tinyint(1), tinyint(2), tinyint(3)), 
    ARRAY(smallint(10), smallint(20), smallint(30), smallint(40)), 
    ARRAY(100, 200, 300), 
    ARRAY(bigint(100000000000000), bigint(20000000000000), bigint(30000000000000), bigint(40000000000000)), 
    ARRAY(true, false, true),
    ARRAY(float(1.23), float(4.56), float(7.89)), 
    ARRAY(double(10.1), double(20.2), double(30.3)), 
    ARRAY("apple", "banana", "orange"),
    ARRAY(TIMESTAMP("2023-07-04 12:00:00"), TIMESTAMP("2023-07-05 12:00:00"), TIMESTAMP("2023-07-06 12:00:00")),
    ARRAY(date("2023-07-04"), date("2023-07-05"), date("2023-07-06"))
),
(
    ARRAY(tinyint(10), tinyint(20), tinyint(30)), 
    ARRAY(smallint(100), smallint(200), smallint(300), smallint(400)), 
    ARRAY(1000, 2000, 3000), 
    ARRAY(bigint(1000000000000000), bigint(200000000000000), bigint(300000000000000), bigint(400000000000000)), 
    ARRAY(true, true,true, true),
    ARRAY(float(12.3), float(45.6), float(78.9)), 
    ARRAY(double(100.1), double(200.2), double(300.3)), 
    ARRAY("abc", "eeee", "sdads"),
    ARRAY(TIMESTAMP("2023-07-02 12:00:00"), TIMESTAMP("2023-07-02 12:00:00"), TIMESTAMP("2023-07-02 12:00:00")),
    ARRAY(date("2021-07-04"), date("2021-07-05"), date("2021-07-06"))
),
(
    ARRAY(tinyint(1), tinyint(2), tinyint(3)), 
    ARRAY(smallint(10), smallint(20), smallint(30), smallint(40)), 
    ARRAY(100, 200, 300), 
    ARRAY(bigint(100000000000000), bigint(20000000000000), bigint(30000000000000), bigint(40000000000000)), 
    ARRAY(true, false, true),
    ARRAY(float(12.3), float(45.6), float(78.9)), 
    ARRAY(double(100.1), double(200.2), double(300.3)), 
    ARRAY("abc", "eeee", "sdads"),
    ARRAY(TIMESTAMP("2023-07-02 12:00:00"), TIMESTAMP("2023-07-02 12:00:00"), TIMESTAMP("2023-07-02 12:00:00")),
    ARRAY(date("2021-07-04"), date("2021-07-05"), date("2021-07-06"))
);

```
Display the results in doris：

```sql
mysql> select * from ywtest.test\G;
*************************** 1. row ***************************
a: [1, 2, 3]
b: [10, 20, 30, 40]
c: [100, 200, 300]
d: [100000000000000, 20000000000000, 30000000000000, 40000000000000]
e: [1, 0, 1]
f: [1.23, 4.56, 7.89]
g: [10.1, 20.2, 30.3]
h: ["apple", "banana", "orange"]
i: [2023-07-04 12:00:00.000000, 2023-07-05 12:00:00.000000, 2023-07-06 12:00:00.000000]
j: [2023-07-04, 2023-07-05, 2023-07-06]
*************************** 2. row ***************************
a: [10, 20, 30]
b: [100, 200, 300, 400]
c: [1000, 2000, 3000]
d: [1000000000000000, 200000000000000, 300000000000000, 400000000000000]
e: [1, 1, 1, 1]
f: [12.3, 45.6, 78.9]
g: [100.1, 200.2, 300.3]
h: ["abc", "eeee", "sdads"]
i: [2023-07-02 12:00:00.000000, 2023-07-02 12:00:00.000000, 2023-07-02 12:00:00.000000]
j: [2021-07-04, 2021-07-05, 2021-07-06]
*************************** 3. row ***************************
a: [1, 2, 3]
b: [10, 20, 30, 40]
c: [100, 200, 300]
d: [100000000000000, 20000000000000, 30000000000000, 40000000000000]
e: [1, 0, 1]
f: [12.3, 45.6, 78.9]
g: [100.1, 200.2, 300.3]
h: ["abc", "eeee", "sdads"]
i: [2023-07-02 12:00:00.000000, 2023-07-02 12:00:00.000000, 2023-07-02 12:00:00.000000]
j: [2021-07-04, 2021-07-05, 2021-07-06]
3 rows in set (0.02 sec)
```
2. Support for specifying delimiters。
```sql
hive> show create table ywtest.cyw14;
OK
CREATE TABLE `ywtest.cyw14`(
  `id` int, 
  `arr` array<array<int>>, 
  `info` int)
ROW FORMAT SERDE 
  'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe' 
WITH SERDEPROPERTIES ( 
  'collection.delim'=',.', 
  'field.delim'='\t', 
  'line.delim'='\n', 
  'serialization.format'='\t') 
STORED AS INPUTFORMAT 
  'org.apache.hadoop.mapred.TextInputFormat' 
OUTPUTFORMAT 
  'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'
LOCATION
  'hdfs://HDFS8000871/usr/hive/warehouse/ywtest.db/cyw14'
TBLPROPERTIES (
  'transient_lastDdlTime'='1688454532')
Time taken: 0.095 seconds, Fetched: 19 row(s)
```
Display the results in doris：
```sql
mysql> select * from ywtest.cyw14;
+------+-----------------------------------+------+
| id   | arr                               | info |
+------+-----------------------------------+------+
|    1 | [[1, 2, 3], [4, 5, 6], [8, 9, 0]] |   11 |
|    2 | [[11, 22, 33], [44, 55, 66]]      |   22 |
+------+-----------------------------------+------+
2 rows in set (0.02 sec)
```

3. Support array multi-level nesting.
Display the results in doris：
```sql
mysql> select * from ywtest.cyw3;
+-----------------------------------------------------------------+----------------------------------------------------+
| info                                                            | str                                                |
+-----------------------------------------------------------------+----------------------------------------------------+
| [[[1], [2, 4]], [[12222, 12313, 123131, 4211], [1, 2], [2, 3]]] | [["hello", "world"], ["hive", "hive", "hivetext"]] |
+-----------------------------------------------------------------+----------------------------------------------------+
1 row in set (0.01 sec)
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/csv/csv_reader.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ CsvReader::CsvReader(RuntimeState* state, RuntimeProfile* profile, ScannerCounte
     _file_compress_type = _params.compress_type;
     _size = _range.size;
 
-    _text_converter.reset(new (std::nothrow) TextConverter('\\'));
+    _text_converter.reset(new (std::nothrow) TextConverter('\\',_array_delimiter[0]));
```
Comment (morningman) [inline be/src/vec/exec/format/csv/csv_reader.cpp:94]:
```suggestion
    _text_converter.reset(new (std::nothrow) TextConverter('\\', _array_delimiter[0]));
```

Comment (morningman) [reply be/src/vec/exec/format/csv/csv_reader.cpp:94]: You can run `sh build-support/clang-format.sh` to format this automatically.

File: be/src/exec/text_converter.h [hunk 2]
```diff
@@ -29,7 +29,7 @@ class SlotDescriptor;
 // numeric types, etc.
 class TextConverter {
 public:
-    TextConverter(char escape_char);
+    TextConverter(char escape_char ,char array_delimiter ='\2');
```
Comment (morningman) [inline be/src/exec/text_converter.h:32]:
```suggestion
    TextConverter(char escape_char, char array_delimiter ='\2');
```

**Comment by morningman** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
> run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Review by kaka11chen — APPROVED** [review]
LGTM

**Comment by hubgeter** [comment]
run buildall

**Review by kaka11chen — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**